src/pages/home/report/ReportActionsView.tsx

```diff
@@ -49,34 +52,160 @@ type ReportActionsViewProps = ReportActionsViewOnyxProps & {
 
     /** The report actions are loading newer data */
     isLoadingNewerReportActions?: boolean;
+
+    /** Whether the report is ready for comment linking */
+    isReadyForCommentLinking?: boolean;
 };
 
+const DIFF_BETWEEN_SCREEN_HEIGHT_AND_LIST = 120;
+const SPACER = 16;
+
+let listIDCount = Math.round(Math.random() * 100);
+
+/**
+ * usePaginatedReportActionList manages the logic for handling a list of messages with pagination and dynamic loading.
+ * It determines the part of the message array to display ('visibleReportActions') based on the current linked message,
+ * and manages pagination through 'handleReportActionPagination' function.
+ *
+ * linkedID - ID of the linked message used for initial focus.
+ * allReportActions - Array of messages.
+ * fetchNewerReportActions - Function to fetch more messages.
+ * route - Current route, used to reset states on route change.
+ * isLoading - Loading state indicator.
+ * triggerListID - Used to trigger a listID change.
+ * returns {object} An object containing the sliced message array, the pagination function,
+ *                   index of the linked message, and a unique list ID.
+ */
+function usePaginatedReportActionList(
+    linkedID: string,
+    allReportActions: OnyxTypes.ReportAction[],
+    fetchNewerReportActions: (newestReportAction: OnyxTypes.ReportAction) => void,
+    route: RouteProp<CentralPaneNavigatorParamList, typeof SCREENS.REPORT>,
+    isLoading: boolean,
+    triggerListID: boolean,
+) {
+    // triggerListID is used when navigating to a chat with messages loaded from LHN. Typically, these include thread actions, task actions, etc. Since these messages aren't the latest, we don't maintain their position and instead trigger a recalculation of their positioning in the list.
+    // we don't set currentReportActionID on initial render as linkedID as it should trigger visibleReportActions after linked message was positioned
+    const [currentReportActionID, setCurrentReportActionID] = useState('');
+    const isFirstLinkedActionRender = useRef(true);
+
+    useLayoutEffect(() => {
+        setCurrentReportActionID('');
+    }, [route]);
+
+    const listID = useMemo(() => {
```

Comment:
Idea: let's create a function like this in `NumberUtils`:

```ts
function generateNewRandomInt(old: number, min: number, max: number): number {
    let newNum = old;
    while (newNum === old) {
        newNum = generateRandomInt(min, max);
    }
    return newNum;
}
```

Then use that to generate the listID with one fewer variable:

```ts
const listID = useMemo(() => NumberUtils.generateNewRandomInt(listID, 1, Number.MAX_SAFE_INTEGER), [
   route,
   isLoadingInitialReportActions,
]);
```

Reply:
I changed it a bit to avoid using an argument within its own computation. That's a nice way to create a save number!